src/pages/home/index.js

```diff
@@ -108,42 +112,82 @@ const Home = () => {
     setLatLng(dataFromChild);
   };
 
+  const switchTag = dataFromChild => {
+    setTabSelected(dataFromChild);
+  };
+
+  const onSubmitEditProfile = data => {
+    editProfile(data);
+  };
+
   return (
     <div className="home">
       <MapView currentPosition={currentPosition} sendLatLng={sendLatLng} targets={targetsList} />
-      <SideBar title={'sideBar.create.title'}>
-        <img className="sidebar__icon" src={sideBarIcon} alt=""></img>
-        <h3 className="sidebar__subtitle">{t('sideBar.subtitle')}</h3>
-        {topics ? (
-          <div>
-            <form className="sidebar__form" onSubmit={handleSubmit(onSubmit)} noValidate>
-              <label htmlFor="radius">{t('home.create.radius')}</label>
-              <Input register={register} type="number" name="radius" />
-
-              <label htmlFor="title">{t('home.create.title')}</label>
-              <Input register={register} type="text" name="title" />
-
-              <label htmlFor="topic_id">{t('home.create.topic')}</label>
-              <Select
-                register={register}
-                name="topic_id"
-                options={topicsList}
-                placeholder={t('home.create.placeholder')}
-              />
-              <div className="saveButton">
-                <Button type="submit">
-                  {isLoading ? t('home.create.savingTarget') : t('home.create.saveTarget')}
-                </Button>
-              </div>
-            </form>
-          </div>
-        ) : null}
+      <SideBar
+        title={tabSelected === 'CREATE_TARGET' ? 'sideBar.create.title' : 'profile.edit.subtitle'}
+        switchTab={switchTag}
+      >
+        {(() => {
+          switch (tabSelected) {
+            case 'CREATE_TARGET':
+              return (
+                <>
+                  <img className="sidebar__icon" src={sideBarIcon} alt="Aim icon"></img>
+                  <h3 className="sidebar__subtitle">{t('sideBar.subtitle')}</h3>
+                  {topics && (
+                    <div>
+                      <form className="sidebar__form" onSubmit={handleSubmit(onSubmit)} noValidate>
```

Comment:
```suggestion
                      <form className="sidebar__form" onSubmit={() => handleSubmit(onSubmit)} noValidate>
```

I think you should pass a function there, otherwise it'd be executed as soon as this form is rendered.

Please check if that's correct before making any changes 😄